packages/react/src/PricingOptions/PricingOptions.tsx

```diff
@@ -0,0 +1,479 @@
+
+import {
+  Accordion,
+  Button,
+  ButtonBaseProps,
+  Heading,
+  HeadingProps,
+  Label,
+  Text,
+  UnorderedList,
+  UnorderedListProps,
+  useWindowSize,
+} from '..'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/pricing-cards/pricing-cards.css'
+
+/**
+ * Main stylesheet (as a CSS Module)
+ */
+import styles from './PricingOptions.module.css'
+
+export type PricingOptionsProps = {
+  presentation?: 'default' | 'cards'
+  ['data-testid']?: string
+} & PropsWithChildren<BaseProps<HTMLDivElement>>
+
+const testIds = {
+  root: 'PricingOptions',
+  item: 'PricingOptions__item',
+  heading: 'PricingOptions__heading',
+  label: 'PricingOptions__label',
+  description: 'PricingOptions__description',
+  price: 'PricingOptions__price',
+  primaryAction: 'PricingOptions__primaryAction',
+  secondaryAction: 'PricingOptions__secondaryAction',
+  featureList: 'PricingOptions__featureList',
+  featureListItem: 'PricingOptions__featureListItem',
+  featureListHeading: 'PricingOptions__featureListHeading',
+  footnote: 'PricingOptions__footnote',
+}
+
+const PricingOptionsRoot = forwardRef(
+  (
+    {
+      children,
+      className,
+      'data-testid': testId,
+      presentation = 'default',
+      ...rest
+    }: PropsWithChildren<PricingOptionsProps>,
+    ref: Ref<HTMLDivElement>,
+  ) => {
+    const filteredChildren = React.Children.toArray(children).filter(
+      child => React.isValidElement(child) && typeof child.type !== 'string' && child.type === PricingOptionsItem,
+    )
+
+    return (
+      <div
+        className={clsx(
+          styles.PricingOptions,
+          styles[`PricingOptions--presentation-${presentation}`],
+          styles[`PricingOptions--items-${filteredChildren.length}`],
+          className,
+        )}
+        data-testid={testId || testIds.root}
+        ref={ref}
+        {...(rest as HTMLAttributes<HTMLElement>)}
+      >
+        {filteredChildren.filter(child => React.isValidElement(child) && child.type === PricingOptionsItem)}
+      </div>
+    )
+  },
+)
+
+export type PricingOptionsItem = {
+  ['data-testid']?: string
+} & PropsWithChildren<BaseProps<HTMLDivElement>>
+
+const PricingOptionsItem = forwardRef(
+  (
+    {animate, 'data-testid': testId, children, className, ...rest}: PropsWithChildren<PricingOptionsItem>,
+    ref: Ref<HTMLDivElement>,
+  ) => {
+    type FilteredChildren = {
+      Content: React.ReactElement<
+        | PricingOptionsLabelProps
+        | PricingOptionsHeadingProps
+        | PricingOptionsDescriptionProps
+        | PricingOptionsPriceProps
+      >[]
+      FeatureList: React.ReactElement<PricingOptionsFeatureListProps> | null
+      Actions: React.ReactElement<PricingOptionsActionsProps>[]
+      Footnote: React.ReactElement<PricingOptionsFootnoteProps> | null
+    }
+
+    const {Content, FeatureList, Actions, Footnote} = React.Children.toArray(children).reduce<FilteredChildren>(
```

Comment:
Nice to have / not a blocker: Given how much is happening here, might be an idea to memoize the value of `React.Children.toArray(children)`.